src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -15,90 +17,48 @@ namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal abstract class AbstractTodoCommentService : ITodoCommentService
     {
-        // we hold onto workspace to make sure given input (Document) belong to right workspace.
-        // since remote host is from workspace service, different workspace can have different expectation
-        // on remote host, so we need to make sure given input always belong to right workspace where
-        // the session belong to.
-        private readonly Workspace _workspace;
-
-        protected AbstractTodoCommentService(Workspace workspace)
-        {
-            _workspace = workspace;
-        }
-
         protected abstract bool PreprocessorHasComment(SyntaxTrivia trivia);
         protected abstract bool IsSingleLineComment(SyntaxTrivia trivia);
         protected abstract bool IsMultilineComment(SyntaxTrivia trivia);
         protected abstract bool IsIdentifierCharacter(char ch);
 
         protected abstract string GetNormalizedText(string message);
         protected abstract int GetCommentStartingIndex(string message);
-        protected abstract void AppendTodoComments(IList<TodoCommentDescriptor> commentDescriptors, SyntacticDocument document, SyntaxTrivia trivia, List<TodoComment> todoList);
-
-        public async Task<IList<TodoComment>> GetTodoCommentsAsync(Document document, IList<TodoCommentDescriptor> commentDescriptors, CancellationToken cancellationToken)
-        {
-            // make sure given input is right one
-            Contract.ThrowIfFalse(_workspace == document.Project.Solution.Workspace);
-
-            // run todo scanner on remote host. 
-            // we only run closed files to make open document to have better responsiveness. 
-            // also we cache everything related to open files anyway, no saving by running
-            // them in remote host
-            if (!document.IsOpen())
-            {
-                var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
```

Comment:
OOP is not handled at the lang-service level.  It now happens entirely at the incremental-analyzer level.  all TODO commetn stuff *only* runs OOP now.

Reply:
This breaks VSMac though, as we use the todo comments in process.

Reply:
Hrmm.  tagging @tmat .  We need to come up with a solution here then.  Roslyn is removing in-process solution crawling entirely.  So this will impact anything you do that needs that.  We've already moved a bunch of services out, and will be intending to get them all out within a sprint or so.  

Do we need a meeting to discuss this?

Reply:
The services don't (shouldn't) really care which process they run in. We could use the `InProcRemoteHostClient` in VS4Mac and run the services in-proc.

Reply:
Currently `InProcRemoteHostClient` is only for testing, but I don't see why we couldn't move it to the product.

Reply:
That works for me. :)